lib/api/api_auth.go

```diff
@@ -82,14 +106,36 @@ func basicAuthAndSessionMiddleware(cookieName string, guiCfg config.GUIConfigura
 			}
 		}
 
+		// Fall back to Basic auth if provided, but don't prompt for it
 		if username, ok := attemptBasicAuth(r, guiCfg, ldapCfg, evLogger); ok {
 			createSession(cookieName, username, guiCfg, evLogger, w, r)
 			next.ServeHTTP(w, r)
+		}
+
+		forbidden(w)
+	})
+
+	handlePasswordLogin := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var req struct{Username string; Password string}
+		err := unmarshalTo(r.Body, &req)
+		if err != nil {
+			l.Debugln("Failed to parse username and password:", err)
+			http.Error(w, "Failed to parse username and password.", 400)
 			return
 		}
 
-		unauthorized(w)
+		if auth(req.Username, req.Password, guiCfg, ldapCfg) {
```

Comment:
I'd prefer to invert the condition

Reply:
See [this other comment](https://github.com/syncthing/syncthing/pull/8757#discussion_r1090902419) for the rationale behind this. Again, I'll make the change here too if you still want me to.

Reply:
As per response to the other comment: Good as is.